include/flatbuffers/stl_emulation.h

```diff
@@ -26,16 +26,17 @@
 #include <memory>
 #include <limits>
 
-// Detect C++17 compatible compiler.
-// __cplusplus >= 201703L - a compiler has support of 'static inline' variables.
-#if defined(FLATBUFFERS_USE_STD_OPTIONAL) \
-    || (defined(__cplusplus) && __cplusplus >= 201703L) \
-    || (defined(_MSVC_LANG) &&  (_MSVC_LANG >= 201703L))
+#ifndef FLATBUFFERS_USE_STD_OPTIONAL
+  // Detect C++17 compatible compiler.
+  // __cplusplus >= 201703L - a compiler has support of 'static inline' variables.
+  #define FLATBUFFERS_USE_STD_OPTIONAL ( \
```

Comment:
Hmm, TIL that it doesn't like definitions with other definitions. Sigh, i hate debugging via CI.

```
#if (defined(__cplusplus) && __cplusplus >= 201703L) \
    || (defined(_MSVC_LANG) && _MSVC_LANG >= 201703L))
   #define FLATBUFFERS_USE_STD_OPTIONAL 1
#endif
```

Reply:
Yes, agreed. Painfully slow.